great_ape_safe/ape_api/ebtc.py

```diff
@@ -866,3 +872,211 @@ def authority_set_authority(self, new_authority):
                 EmptyBytes32,
                 delay + 1,
             )
+
+    #### ===== CDP OPS ===== ####
+
+    def _assert_collateral_balance(self, coll_amount):
+        total_coll_bal = self.collateral.balanceOf(self.safe.address)
+        assert total_coll_bal >= self.collateral.getSharesByPooledEth(coll_amount)
+
+    def _assert_debt_balance(self, debt_amount):
+        total_debt_bal = self.ebtc_token.balanceOf(self.safe.address)
+        assert total_debt_bal >= debt_amount
+        return total_debt_bal
+
+    def _assert_cdp_id_ownership(self, cdp_id):
+        cdp_safe_owned = self.sorted_cdps.getCdpsOf(self.safe.address)
+        assert cdp_id in cdp_safe_owned
+
+    def _hint_helper_values(self, coll_amount, debt_amount):
+        nicr = self.hint_helpers.computeNominalCR(coll_amount, debt_amount)
+
+        total_cdps = self.sorted_cdps.getSize()
+        hint, _, _ = self.hint_helpers.getApproxHint(
+            nicr, total_cdps, 42
+        )  # random seed 42
+
+        upper_hint, lower_hint = self.sorted_cdps.findInsertPosition(nicr, hint, hint)
+
+        return upper_hint, lower_hint
+
+    def open_cdp(self, coll_amount, target_cr):
+        """
+        @dev Opens a new cdp position.
+        @param coll_amount The total stETH collateral amount deposited for the specified Cdp.
+        @param target_cr The desired target collateral ratio in the cdp position.
```

Comment:
I would add details around the format and decimals, don't think it is intuitive that the CR is expected in 18 decimal wei format.

Reply:
hints suggestions included in d2ae0dafaea94b24d897967c623681365849f29c